pandas/_libs/tslibs/offsets.pyx

```diff
@@ -304,6 +304,15 @@ class _BaseOffset(object):
     _day_opt = None
     _attributes = frozenset(['n', 'normalize'])
 
+    def __init__(self, n=1, normalize=False):
+        n = self._validate_n(n)
+        object.__setattr__(self, "n", n)
```

Comment:
I would just set these as regular attributes, I doubt this actually makes a signfiicant perf difference.

Reply:
Not sure I understand.  These _are_ regular attributes, but the PR overrides `__setattr__` so we have to use `object.__setattr__`.

Reply:
oh sorry you are right. 

can you set these as cdef readonly? (and maybe type them) or not yet?

Reply:
I really really wish I could, but until someone smarter than me looks at #18824 `cdef class` causes pickle test failures.

Reply:
@jbrockmendel what pickle issue are you referring to in that change? That just looks like a master tracker but not sure the exact issue contained within

Reply:
@WillAyd thanks for catching that, should be #18224.  Will fix above.